frame/oracle/src/lib.rs

```diff
@@ -163,7 +169,7 @@ pub mod pallet {
 		type MaxPrePrices: Get<u32>;
 
 		/// The weight information of this pallet.
-		type WeightInfo: WeightInfo;
+		type OracleWeightInfo: WeightInfo;
```

Comment:
Not sure why is renamed neeed? Any weight info is of pallet. If you want to share configs may be look into DeFiComposableConfig example

Reply:
>1.<T as pallet_oracle:Config>::WeigthInfo on call side
1.1. pallet_timestamp::Config do you use WeigthInfo from that trait? feels like bad smell.
1.2 i seen never such was done in any pallet and there are option to do diffently

I don't use the `WeightInfo` from that trait but there is a conflict since they both have the same name. I don't think there are other pallets extending  `pallet_timestamp::Config` trait to access it's storage.

>3. let slot_duration: u64 = <T as pallet_timestamp::Config>::MinimumPeriod::get. You can add type MinimumPeriod into pallet_oracle::Config and init it in runtime.

I considered this, but then the code would allow changing the pallet_oracle::Config without the actual `MinimumPeriod` of the chain having changed. The way I use this now is the way aura and babe use it refer, https://github.com/paritytech/substrate/blob/f318350b6410ef47e438aae24c0b4779373d0a7b/frame/babe/src/lib.rs#L468

Not sure about 3.1 and 3.2, let me check